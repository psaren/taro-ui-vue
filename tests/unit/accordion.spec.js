import { mount } from '@vue/test-utils'
import AtAccordion from '../../src/components/accordion'

const factory = (values = {}, slots = { default: ['按钮'] }) => {
  return mount(AtAccordion, {
    slots,
    propsData: { ...values },
  })
}

describe('AtAccordion Snap', () => {
  it('render initial AtAccordion', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props title', () => {
    const wrapper = factory({
      title: 'title',
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props open', () => {
    const wrapper = factory({
      open: true,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props icon', () => {
    const wrapper = factory(
      {
        icon: { value: 'chevron-down', color: 'red' },
      },
      {
        default: ['<view></view>'],
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props icon prefixClass', () => {
    const wrapper = factory(
      {
        icon: { prefixClass: 'prefixClass', value: 'star', color: 'red' },
      },
      {
        default: ['<view></view>'],
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtAccordion -- props note', () => {
    const wrapper = factory(
      {
        note: 'note',
      },
      {
        default: ['<view></view>'],
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtAccordion Event', () => {
  it('toggle accordion', () => {
    const onClick = jest.fn()
    const wrapper = factory({
      onClick: onClick,
    })
    wrapper.find('.at-accordion__header').trigger('tap')
    expect(onClick).toBeCalled()
  })
})
