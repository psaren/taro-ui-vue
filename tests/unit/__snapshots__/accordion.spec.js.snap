// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtAccordion Snap render AtAccordion -- props icon 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="at-icon at-icon-chevron-down at-accordion__icon"
      style="color: red;"
    />
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      />
      <view
        class="at-accordion__info__note"
      />
    </view>
    <view
      class="at-accordion__arrow"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content at-accordion__content--inactive"
  >
    <view
      class="at-accordion__body"
    >
      <view />
    </view>
  </view>
</view>
`;

exports[`AtAccordion Snap render AtAccordion -- props icon prefixClass 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="prefixClass prefixClass-star at-accordion__icon"
      style="color: red;"
    />
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      />
      <view
        class="at-accordion__info__note"
      />
    </view>
    <view
      class="at-accordion__arrow"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content at-accordion__content--inactive"
  >
    <view
      class="at-accordion__body"
    >
      <view />
    </view>
  </view>
</view>
`;

exports[`AtAccordion Snap render AtAccordion -- props note 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      />
      <view
        class="at-accordion__info__note"
      >
        note
      </view>
    </view>
    <view
      class="at-accordion__arrow"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content at-accordion__content--inactive"
  >
    <view
      class="at-accordion__body"
    >
      <view />
    </view>
  </view>
</view>
`;

exports[`AtAccordion Snap render AtAccordion -- props open 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      />
      <view
        class="at-accordion__info__note"
      />
    </view>
    <view
      class="at-accordion__arrow at-accordion__arrow--folded"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content"
  >
    <view
      class="at-accordion__body"
    >
      按钮
    </view>
  </view>
</view>
`;

exports[`AtAccordion Snap render AtAccordion -- props title 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      >
        title
      </view>
      <view
        class="at-accordion__info__note"
      />
    </view>
    <view
      class="at-accordion__arrow"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content at-accordion__content--inactive"
  >
    <view
      class="at-accordion__body"
    >
      按钮
    </view>
  </view>
</view>
`;

exports[`AtAccordion Snap render initial AtAccordion 1`] = `
<view
  class="at-accordion"
>
  <view
    class="at-accordion__header"
  >
    <view
      class="at-accordion__info"
    >
      <view
        class="at-accordion__info__title"
      />
      <view
        class="at-accordion__info__note"
      />
    </view>
    <view
      class="at-accordion__arrow"
    >
      <view
        class="at-icon at-icon-chevron-down"
      />
    </view>
  </view>
  <view
    class="at-accordion__content at-accordion__content--inactive"
  >
    <view
      class="at-accordion__body"
    >
      按钮
    </view>
  </view>
</view>
`;
